src/lib/mdns/TxtFields.cpp

```diff
@@ -56,34 +57,35 @@ bool IsKey(const ByteSpan & key, const char * desired)
     return true;
 }
 
-uint16_t MakeU16FromAsciiDecimal(const ByteSpan & val)
+uint32_t MakeU32FromAsciiDecimal(const ByteSpan & val)
 {
-    // Largest u16 number if 65536, so 5 chars.
-    constexpr size_t kMaxUint16StrLen = 5;
-    // Add a space for null
-    char temp[kMaxUint16StrLen + 1];
-    if (val.size() > kMaxUint16StrLen)
-    {
-        return 0;
-    }
-    Platform::CopyString(temp, sizeof(temp), val);
-    char * end;
-    unsigned long num = strtoul(temp, &end, 10);
-    if (num > std::numeric_limits<uint16_t>::max())
+    char nullTerminatedValue[std::numeric_limits<uint32_t>::digits10 + 2];
+
+    if (val.size() >= sizeof(nullTerminatedValue))
     {
         return 0;
     }
-    return static_cast<uint16_t>(num);
+
+    Platform::CopyString(nullTerminatedValue, sizeof(nullTerminatedValue), val);
+    unsigned long num = strtoul(nullTerminatedValue, nullptr, 10);
```

Comment:
Shouldn't we have better handling of non-digit chars in the value here?  Per spec those should presumably lead to the value being ignored.....

Reply:
Right, added more checks.